src/test/java/core/basesyntax/FruitServiceImplTest.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax;
+
+import core.basesyntax.database.Storage;
+import core.basesyntax.dto.FruitRecordDto;
+import core.basesyntax.handler.DecreaseOperationHandler;
+import core.basesyntax.handler.IncreaseOperationHandler;
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.implementions.FruitServiceImpl;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitServiceImplTest {
+    private static FruitService fruitService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<OperationType, OperationHandler> operationStrategyMap = new HashMap<>();
+        OperationHandler increaseHandler = new IncreaseOperationHandler();
+
+        operationStrategyMap.put(OperationType.BALANCE, increaseHandler);
+        operationStrategyMap.put(OperationType.RETURN, increaseHandler);
+        operationStrategyMap.put(OperationType.SUPPLY, increaseHandler);
+        operationStrategyMap.put(OperationType.PURCHASE, new DecreaseOperationHandler());
+
+        fruitService = new FruitServiceImpl(new OperationStrategyImpl(operationStrategyMap),
+                                            new FruitRecordDto());
+    }
+
+    @Test
+    public void saveData_chooseCorrectStrategy_isOk() {
+        Fruit apple = new Fruit("apple");
+        Fruit banana = new Fruit("banana");
+        List<FruitRecordDto> recordDtos = List.of(
```

Comment:
What is this Dtos? Maybe better just recordDto

Reply:
done